data/mods/gen2stadium/scripts.ts

```diff
@@ -0,0 +1,555 @@
+/**
+ * Stadium 2 mechanics inherit from gen 2 mechanics, but fixes some bugs. 
+ */
+export const Scripts: ModdedBattleScriptsData = {
+	inherit: 'gen2',
+	gen: 2,
+	pokemon: {
+		inherit: true,
+		getStat(statName, unboosted, unmodified, fastReturn) {
+			// @ts-ignore - type checking prevents 'hp' from being passed, but we're paranoid
+			if (statName === 'hp') throw new Error("Please read `maxhp` directly");
+
+			// base stat
+			let stat = this.storedStats[statName];
+
+			// Stat boosts.
+			if (!unboosted) {
+				let boost = this.boosts[statName];
+				if (boost > 6) boost = 6;
+				if (boost < -6) boost = -6;
+				if (boost >= 0) {
+					const boostTable = [1, 1.5, 2, 2.5, 3, 3.5, 4];
+					stat = Math.floor(stat * boostTable[boost]);
+				} else {
+					const numerators = [100, 66, 50, 40, 33, 28, 25];
+					stat = Math.floor(stat * numerators[-boost] / 100);
+				}
+			}
+			if (this.status === 'par' && statName === 'spe' && this.volatiles['parspeeddrop']) {
+				stat = Math.floor(stat / 4);
+			}
+			if (!unmodified) {
+				if (this.status === 'brn' && statName === 'atk' && this.volatiles['brnattackdrop']) {
+					stat = Math.floor(stat / 2);
+				}
+			}
+
+			// Gen 2 caps stats at 999 and min is 1.
+			stat = this.battle.clampIntRange(stat, 1, 999);
+			if (fastReturn) return stat;
+
+			// Screens
+			if (!unboosted) {
+				if (
+					(statName === 'def' && this.side.sideConditions['reflect']) ||
+					(statName === 'spd' && this.side.sideConditions['lightscreen'])
+				) {
+					stat *= 2;
+				}
+			}
+
+			// Handle boosting items
+			if (
+				(['Cubone', 'Marowak'].includes(this.species.name) && this.item === 'thickclub' && statName === 'atk') ||
+				(this.species.name === 'Pikachu' && this.item === 'lightball' && statName === 'spa')
+			) {
+				stat *= 2;
+			} else if (this.species.name === 'Ditto' && this.item === 'metalpowder' && ['def', 'spd'].includes(statName)) {
+				stat *= 1.5;
+			}
+
+			return stat;
+		},
+	},
+	// Stadium 2 shares gen 2 code but it fixes some problems with it.
+	actions: {
+		inherit: true,
+		tryMoveHit(target, pokemon, move) {
+			const positiveBoostTable = [1, 1.33, 1.66, 2, 2.33, 2.66, 3];
+			const negativeBoostTable = [1, 0.75, 0.6, 0.5, 0.43, 0.36, 0.33];
+			const doSelfDestruct = true;
+			let damage: number | false | undefined = 0;
+
+			if (move.selfdestruct && doSelfDestruct) {
+				this.battle.faint(pokemon, pokemon, move);
+				/**
+				 * Keeping track of the last move used for self-ko clause, Making sure to clear the opponents last move so that self-destruct and explosion
+				 * does not persist between Pokemon, preventing problems caused by situations such as a player from blowing up both they and their opponents second last Pokemon
```

Comment:
```suggestion
				 * Keeping track of the last move used for self-ko clause,
				 * making sure to clear the opponents last move so that self-destruct and explosion
				 * does not persist between Pokemon, preventing problems caused by situations,
				 * such as a player from blowing up both they and their opponents second last Pokemon
```